prov/util/src/util_mem_monitor.c

```diff
@@ -194,6 +194,21 @@ int ofi_monitors_add_cache(struct ofi_mem_monitor **monitors,
 			ret = monitor->start(monitor);
 			if (ret) {
 				cache->monitors[iface] = NULL;
+
+				/* Requested memory monitor is not supported.
+				 * The cache cannot be used for this HMEM type.
+				 */
+				if (ret == -FI_ENOSYS) {
+					FI_DBG(&core_prov, FI_LOG_MR,
+					       "MR cache not supported for %s memory\n",
+					       fi_tostr(&iface, FI_TYPE_HMEM_IFACE));
+					ret = FI_SUCCESS;
+					continue;
+				}
+
+				FI_WARN(&core_prov, FI_LOG_MR,
+					"MR cache disabled for %s memory\n",
+					fi_tostr(&iface, FI_TYPE_HMEM_IFACE));
 				goto out;
```

Comment:
This is the error case.  We need to decide the proper way to handle an error here.  Either we simply skip the monitor, log an error, and continue; or we abort completely.  In the latter case, we need to undo the monitors that were added successfully.

The current (and proposed) code wants to abort, but doesn't clean up.  To keep the current behavior, change this to a goto err case which handles the cleanup and returns the failure.  Otherwise restructure the code to always continue in case of an error

Reply:
I think aborting makes since for this error case since the user has request a valid memory monitor and it could not be started. I'll update code to undo during abort.